appendices/assignments.md

```diff
@@ -0,0 +1,199 @@
+# Course Assignments
+
+<!-- Perspective for this section would be on the student, but the instructor
+section would be of course on the instructor. Or should this be entirely 
+targeted to the instructor? -->
+At the end of this book, you will have written at least one R package. 
+To help you get there, we provide "assignments" --- 
+projects we expect you complete as you move through the book.
+You would largely work on these assignments outside of class time.
+Each project assignment has several milestones you would need to complete
+throughout the course, usually due shortly after encountering them during
+the course. The project assignments are to:
+
+1. Create a small and simple package on your own.
+2. Make small and simple improvements to a package on GitHub.
+3. Create a slightly larger, team-based package that you would complete in a
+group of 2-3 other learners. This last assignment is optional and depends on
+the specific context of the course.
+
+Each project assignment is described in more detail below.
+
+## First project: Create a package on your own
+
+The first assignment is to create a small, simple package. The *learning
+objective* for this project is to practice and reinforce the skills 
+and knowledge you've learned during the course and to start building your
+confidence in package development that will help for the later assignments.
+
+<!--
+NOTE: I moved this due date to after chapter 13 rather than chapter 7, since
+we could keep writing the final exercises to keep building up this package.
+Plus, for those learners who will use this on their own, this would be nice to
+keep going with.
+-->
+
+The project assignment is recommended to be **due** immediately after the
+Package Release lesson (Chapter \@ref(package-release)). At the end of each
+chapter there is a final exercise that gives you tasks to do that guide you in
+completing this first project assignment. These final exercises are used as 
+milestones to ensure you are making progress on your assignment.
+
+An *acceptable* package for completing the assignment would have at a *minimum*:
+
+- Two-three functions.
+- (Optional) a dataset with associated roxygen documentation and an R script that
```

Comment:
Given the intro to these requirements states these are a _minimum_, should we split out the things that are optional? Are there other things that could be listed as optional?